routers/private/hook_pre_receive.go

```diff
@@ -124,6 +124,10 @@ func HookPreReceive(ctx *gitea_context.PrivateContext) {
 			preReceiveTag(ourCtx, refFullName)
 		case git.DefaultFeatures().SupportProcReceive && refFullName.IsFor():
 			preReceiveFor(ourCtx, refFullName)
+		case refFullName.IsPull():
+			ctx.JSON(http.StatusForbidden, private.Response{
+				UserMsg: "Can't update pull request manually.",
+			})
```

Comment:
Wrong. You are dong `ctx.JSON` in a `for` loop.

(and: no description, no comment, no test)

Reply:
> It will return after the following

Updated my comment, I meant: ideally it shouldn't "block other operations", unless there is a strong reason.

Reply:
I think you are right. These update operations should be ignored. So this is just a quick patch to prevent possible wrong deletion.